backend/maelstro/main.py

```diff
@@ -100,6 +100,14 @@ def get_sources() -> list[dict[str, str]]:
 def get_destinations() -> list[dict[str, str]]:
     return config.get_destinations()
 
+@app.post("/search/{src_name}")
+def post_search(src_name: str, data: Dict[str, Any]) :
+    src_info = config.get_access_info(
+        is_src=True, is_geonetwork=True, instance_id=src_name
+    )
+    gn = GnApi(src_info["url"], src_info["auth"])
+
+    return gn.search(json.dumps(data))
```

Comment:
you can directluse the json() junfion of the Response object instead of json.dumps()
see here:
https://github.com/camptocamp/python-geonetwork/blob/82e3b49a80512e64e826947da3f2e79ab5cb756b/geonetwork/gn_api.py#L91